llvm/lib/Target/AMDGPU/AMDGPUPromoteAlloca.cpp

```diff
@@ -64,10 +66,21 @@ static cl::opt<unsigned> PromoteAllocaToVectorLimit(
     cl::desc("Maximum byte size to consider promote alloca to vector"),
     cl::init(0));
 
+static cl::opt<bool> PromoteAllocaWholeFn(
+    "promote-alloca-vector-whole-function",
```

Comment:
I don't think this should be an option. It should simply be the behavior. There was nothing desirable about the random ordering behavior as before